API-GUIDELINE.md

```diff
@@ -0,0 +1,254 @@
+             "type": "centPrecision",
+             "fractionDigits": 2,
+             "currencyCode": "EUR",
+             "centAmount": 5600
+           }
+         }
+       ],
+       "store": {
+         "typeId": "store",
+         "key": "test-key"
+       },
+       "cartState": "Active",
+       "totalPrice": {
+         "type": "centPrecision",
+         "fractionDigits": 2,
+         "currencyCode": "EUR",
+         "centAmount": 5600
+       },
+       "customLineItems": [],
+       "discountCodes": [],
+       "inventoryMode": "None",
+       "taxMode": "Platform",
+       "taxRoundingMode": "HalfEven",
+       "taxCalculationMode": "LineItemLevel",
+       "refusedGifts": [],
+       "origin": "Customer"
+     }
+
+## Naming convention
+
+### General Guidelines
+To use consistent names these are the rules to follow:
+
+ DO'S
+ - use nouns instead of verbs to expose resources
+ - use lowercase and separated by hyphens (-) for path segments like /shopping-lists
+ - use American English
+ - create each resource as unique resource name
+ - use the proper HTTP methods for API operations whenever possible
+ 
+ DON'TS
+ - DON'T use snake case (_) or other punctuation
+ - DON'T use file extensions
+ - DON'T use abbreviations
+ - DON'T use CRUD HTTP methods or verbs
+
+###Naming Properties
+The properties are identified in base of the type: 
+ - DateTime
+ - Boolean
+ - Number/Integer
+ - String
+ - Array
+ - Object
+
+#### DateTime
+In general, define a DateTime property using a name ending with "At", then if there is a range of date use "From" and "To".
+
+Some examples:
+
+        returnAt
+        expirationAt
+        validFrom
+        validTo
+
+#### Boolean
+Define a Boolean property using a name which NOT includes "is" but it can include "has" and the verb at past participle.
+
+Some examples:
+
+        emailVerified
+        hasStagedChanges
+        published
+        active
+        
+#### Number/Integer
+Define a Number/Integer property using names that can be quantified.
+
+Some examples:
+
+        availableQuantity
+        restockableInDays
+        version
+        weight
+        height
+        variantId
+        totalAmount
+      
+#### String
+Define a String property using names that have to be explicable about the content.
+
+Some examples:
+
+      orderEditResult
+      productName
+      productKey
+      productSku
+      discountCode
+         
+#### Array
+Define an Array property using names in plural form, typically ending with "s".
+
+Some examples:
+
+    discountCodes
+    refusedGifts
+    roles
+    allowedValues
+    
+#### Object
+Define an Object property using names that have to be explicable about the content and the scope.
+
+Some examples:
+
+    shippingMethod
+    shippingRateInput
+    product
+    shippingDetailsToRemove
+    discountedPricePerQuantity
+    discountedPrice 
+
+#### Exceptions
+The only exception allowed is:
+ - write "id" even if is an abbreviation
```

Comment:
Let's add some paragraph about method names to this. Create methods are pretty straightforward, I think: "CreateResource"

For update actions we still hold onto this convention?
**set**Field: for optional properties, can be unset if value is empty
**change**Field: for mandatory fields, value cannot be empty

Retrieve methods:
**Query Resources** returns PagedQueryResult containing multiple resources (plural)
**Get Resource by Unique Identifier**, returns only the identified Resource (singular)

Unique Identifier can be UUID, Key, customer email, etc.

Reply:
please follow further communication here https://github.com/commercetools/api-design-guidelines/issues/4